src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field23.java

```diff
@@ -0,0 +1,136 @@
+ *
+ * @see <a href="https://www2.swift.com/knowledgecentre/publications/usgf_20230720/2.0?topic=idx_fld_tag_23B.htm">
+ * Field 23B</a>
+ * @see <a href="https://www2.swift.com/knowledgecentre/publications/usgf_20230720/2.0?topic=idx_fld_tag_23E.htm">
+ * Field 23E</a>
+ */
+public class Field23 {
+
+    public static final String TAG = "23";
+
+    /**
+     * List of options with current implementation.
+     */
+    private static final List<Character> OPTIONS = Arrays.asList(
+            ConnectorConstants.OPTION_B, ConnectorConstants.OPTION_E);
+    private static final Map<Character, String> FIELD_NAME = Map.of(
+            ConnectorConstants.OPTION_B, MT103Constants.BANK_OPERATION_CODE,
+            ConnectorConstants.OPTION_E, MT103Constants.INSTRUCTION_CODE
+    );
+    private char option;
+
+    // example: TELI
+    private String code;
+
+    // example: 3226553478
+    private String additionalInformation;
+
+    /**
+     * Constructor to initialize all attributes.
+     *
+     * @param option Single character
+     * @param code String with four uppercase letters
+     * @param additionalInformation Optional string with character set x
+     */
+    public Field23(char option, String code, String additionalInformation) {
+        this.option = option;
+        this.code = code;
+        this.additionalInformation = additionalInformation;
+    }
+
+    public char getOption() {
+        return option;
+    }
+
+    public void setOption(char option) {
+        this.option = option;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public void setCode(String code) {
+        this.code = code;
+    }
+
+    public String getAdditionalInformation() {
+        return additionalInformation;
+    }
+
+    public void setAdditionalInformation(String additionalInformation) {
+        this.additionalInformation = additionalInformation;
+    }
+
+    /**
+     * Method to parse and get Field23 object.
+     * Current Implementations -> Option B and E
+     *
+     * @param field23String String containing value of 23 field in Text Block
+     * @param option        single character option of the field23String
+     * @return An instance of this model.
+     * @throws MTMessageParsingException if the value is invalid
+     */
+    public static Field23 parse(String field23String, char option) throws MTMessageParsingException {
+
+        if (!OPTIONS.contains(option)) {
+            throw new MTMessageParsingException(String.format(MTParserConstants.INVALID_OPTION_FOR_FIELD, option,
+                    ConnectorConstants.FIELD_23));
+        }
+
+        Matcher field23Matcher = MTParserConstants.FIELD_23_REGEX_PATTERN.matcher(field23String);
+
+        if (field23Matcher.matches()) {
+
+            // group 1 -> Code
+            // group 2 -> /Additional Information
+            // group 3 -> Additional Information
+            return new Field23(option, field23Matcher.group(1), field23Matcher.group(3));
+
```

Comment:
shall we remove these extra lines

Reply:
Remove in commit 7b56b383e3127da22c1b7305f9f84c495346f0f3.